### PR TITLE
Fix Critical damage

### LINKED ISSUE #710: Imbuement Bugs
Imbuement:
a)Não dando critico no fisico(meele/distance single target, arrows,etc).
b) Tempo do imbuement contando mesmo dentro do depot + Tempo do Imbuement contando na backpack
(Deveria contar apenas com CONDITION_INFIGHT e ao perder essa condition, parar de decair o tempo)
c)Itens que já foram imbuidos anteriormente, mesmo com o imbuement removido,NÃO SÃO negociáveis no market.

ps : se alguém conhecer algum outro bug nos imbuiments, por favor, publicar a baixo =)

### PR DESCRIPTION
**Pull request by lBaah**
Fix melee critical damage on health.
Fix critical damage on players with manashield.
Fix #710 a) and #694 
 

Pode mandar a minha parte, Monobraço Costa

### CODE REVIEW

File: src/combat.cpp [hunk 1]
```diff
@@ -516,9 +530,13 @@ void Combat::CombatManaFunc(Creature* caster, Creature* target, const CombatPara
 {
 	assert(data);
 	CombatDamage damage = *data;
-	if (damage.primary.value < 0) {
-		if (caster && caster->getPlayer() && target->getPlayer()) {
-			damage.primary.value /= 2;
+	if ((damage.primary.value < 0 || damage.secondary.value < 0) && caster) {
```
Comment (gerotib) [inline src/combat.cpp]: I tested and manadrain spells used by monsters caused server crash. With 
```suggestion
	if ((damage.primary.value < 0 || damage.secondary.value < 0)) {
		if (caster && caster->getPlayer() && target->getPlayer()) {
```
as before it seems to work fine.

**Comment by gerotib** [comment]
And also players in pvp are getting hit for 100% of dmg + halfed critical dmg (it should be applied first and then halfed for normal players, and not being halfed for players with bs). Heres my quick fix that probably could be simplified.
EDIT: simplified already, in `else if` check if caster is player is needed, rest I dont think so.
```cpp
	if ((damage.primary.value < 0 || damage.secondary.value < 0) && caster) {
		Player* targetPlayer = target->getPlayer();
		if (targetPlayer && caster->getPlayer() && targetPlayer->getSkull() != SKULL_BLACK) { // player w/o black skull is getting hit for 50%
			// Critical dmg section
			uint16_t chance = caster->getPlayer()->getSkillLevel(SKILL_CRITICAL_HIT_CHANCE);
			if (chance != 0 && uniform_random(1, 100) <= chance) {
				damage.critical = true;
				damage.primary.value += ((damage.primary.value * caster->getPlayer()->getSkillLevel(SKILL_CRITICAL_HIT_DAMAGE)) / 100); //add critical dmg to primary dmg
				damage.secondary.value += ((damage.secondary.value * caster->getPlayer()->getSkillLevel(SKILL_CRITICAL_HIT_DAMAGE)) / 100); //add critical dmg to secondary dmg
			}
			damage.primary.value /= 2; // half the primary damage in pvp
			damage.secondary.value /= 2; // half the secondary damage in pvp
		}
		else if (caster->getPlayer()) { // player with black skull and monsters are getting hit for 100% (not being halfed)
			// Critical dmg section
			uint16_t chance = caster->getPlayer()->getSkillLevel(SKILL_CRITICAL_HIT_CHANCE);
			if (chance != 0 && uniform_random(1, 100) <= chance) {
				damage.critical = true;
				damage.primary.value += ((damage.primary.value * caster->getPlayer()->getSkillLevel(SKILL_CRITICAL_HIT_DAMAGE)) / 100);
				damage.secondary.value += ((damage.secondary.value * caster->getPlayer()->getSkillLevel(SKILL_CRITICAL_HIT_DAMAGE)) / 100);
			}
		}
	}
```

**Comment by lBaah** [comment]
@gerotib Great review as always.
I am reviewing the code since I was too tired the day I made these changes.

So we have to half the damage for mana also.

File: src/combat.cpp [hunk 2]
```diff
@@ -516,9 +530,14 @@ void Combat::CombatManaFunc(Creature* caster, Creature* target, const CombatPara
...
-	if (damage.primary.value < 0) {
+	if ((damage.primary.value < 0 || damage.secondary.value < 0)) {
 		if (caster && caster->getPlayer() && target->getPlayer()) {
-			damage.primary.value /= 2;
+		// Critical damage
+		uint16_t chance = caster->getPlayer()->getSkillLevel(SKILL_CRITICAL_HIT_CHANCE);
+		if (chance != 0 && uniform_random(1, 100) <= chance) {
+			damage.critical = true;
+			damage.primary.value += ((damage.primary.value * caster->getPlayer()->getSkillLevel(SKILL_CRITICAL_HIT_DAMAGE)) / 100)/2;
+			damage.secondary.value += ((damage.secondary.value * caster->getPlayer()->getSkillLevel(SKILL_CRITICAL_HIT_DAMAGE)) / 100)/2;
```
Comment (gerotib) [inline src/combat.cpp]:
```suggestion
			damage.secondary.value += ((damage.secondary.value * caster->getPlayer()->getSkillLevel(SKILL_CRITICAL_HIT_DAMAGE)) / 100)/2;
			}
```
missing "}"

**Comment by gerotib** [comment]
> So we have to half the damage for mana also.

As I tested both ways it works fine

**Comment by lBaah** [comment]
@gerotib travis now likes it. Can you check damage on mana again?

**Comment by gerotib** [comment]
@lBaah everything seems to work fine

**Comment by majestyotbr** [comment]
LGTM

**Pull request merged**